lib/srv/db/cloud/watchers/watcher.go

```diff
@@ -119,11 +119,11 @@ func (w *Watcher) fetchAndSend() {
 	for _, fetcher := range w.fetchers {
 		databases, err := fetcher.Get(w.ctx)
 		if err != nil {
-			w.log.WithError(err).Errorf("%s failed.", fetcher)
-			return
+			w.log.WithError(err).Warnf("%s failed.", fetcher)
 		}
 		result = append(result, databases...)
 	}
+
```

Comment:
I think empty slice is a valid condition. but, not sure if there is some certain cases we don't want to send at all?

Reply:
Yeah, I thought about this too. If we just ignore errors here, the following scenario could happen:

1. Say, one of the fetchers fails due to an AWS issue in e.g. `us-east-1` (case in point, happened today).
2. We send a resulting list of databases which won't include `us-east-1` databases to the [watcher](https://github.com/gravitational/teleport/blob/v8.0.6/lib/srv/db/watcher.go#L121-L127).
3. Watcher then [reconciles](https://github.com/gravitational/teleport/blob/v8.0.6/lib/srv/db/watcher.go#L49-L54) them with its list of proxied databases and stops proxying all `us-east-1` ones cause they're missing from the newly fetched list (so from it's perspective they were removed).

I think we need to tweak the fetch/reconcile logic to not send an empty list in case of unexpected errors to avoid this. WDYT?

Reply:
yeah. In one version of changes i have something like `if hasError && result.Len() == 0` then don't send. Though, what if the error is unrecoverable in this case (say devop revoked all our iam permission)? I think we will stuck with old list forever. I also feel the behaviour is a little inconsistent this way as: in a partial failure, we remove some resources, vs, in a full failure, we keep everything as before.

If we do do this though, maybe we should very old existing resources in reconciles (use the expiry?)